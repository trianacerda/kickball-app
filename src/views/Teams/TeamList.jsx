import { useEffect, useState } from 'react';
import { NavLink, Link } from 'react-router-dom';
import { getTeams } from '../../services/teams';

function TeamList() {
  const [teams, setTeams] = useState([]);

  useEffect(() => {
    getTeams().then((resp) => setTeams(resp));
  }, []);

  return (
    <>
      <NavLink to="/" className="home-link" exact>
        Home
      </NavLink>
      <NavLink to="/players" className="player-link" exact>
        Players
      </NavLink>

      <h1>List of Teams:</h1>
      <ul>
        {teams.map((team) => {
          return (
            <li key={team.id}>
              <Link to={`/teams/${team.id}`} className="team-list-link">
                {team.name}
              </Link>
            </li>
          );
        })}
      </ul>
    </>
  );
}

export default TeamList;
