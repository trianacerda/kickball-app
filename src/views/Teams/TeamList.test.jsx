import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import TeamList from './TeamList';

it('should render a list of teams', async () => {
  const { container } = render(
    <MemoryRouter>
      <TeamList />
    </MemoryRouter>
  );

  screen.getByText('Loading teams.....');
  const team1 = await screen.findByText('Tala Loves Snacks');
  const teamLabel = await screen.findByRole('list', { name: 'teams' });

  expect(team1).toBeInTheDocument();
  expect(teamLabel).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
