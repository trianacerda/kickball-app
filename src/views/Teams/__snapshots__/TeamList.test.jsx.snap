// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of teams 1`] = `
<div>
  <a
    class="home-link"
    href="/"
  >
    Home
  </a>
  <a
    class="player-link"
    href="/players"
  >
    Players
  </a>
  <a
    class="new-team-link"
    href="/teams/new"
  >
    Add a New Team
  </a>
  <h1>
    List of Teams:
  </h1>
  <ul
    aria-label="teams"
  >
    <li>
      <a
        class="team-list-link"
        href="/teams/3"
        name="Luna's Looney Tunes"
      >
        Luna's Looney Tunes
      </a>
      <a
        class="update-team-link"
        href="/teams/3/update"
      >
        Update a Team
      </a>
      <button
        aria-label="DeleteLuna's Looney Tunes"
        type="button"
        value="delete"
      >
        Delete
      </button>
    </li>
    <li>
      <a
        class="team-list-link"
        href="/teams/7"
        name="Mister loves food"
      >
        Mister loves food
      </a>
      <a
        class="update-team-link"
        href="/teams/7/update"
      >
        Update a Team
      </a>
      <button
        aria-label="DeleteMister loves food"
        type="button"
        value="delete"
      >
        Delete
      </button>
    </li>
    <li>
      <a
        class="team-list-link"
        href="/teams/1"
        name="Tala Loves Snacks"
      >
        Tala Loves Snacks
      </a>
      <a
        class="update-team-link"
        href="/teams/1/update"
      >
        Update a Team
      </a>
      <button
        aria-label="DeleteTala Loves Snacks"
        type="button"
        value="delete"
      >
        Delete
      </button>
    </li>
    <li>
      <a
        class="team-list-link"
        href="/teams/2"
        name="Tony as Monk"
      >
        Tony as Monk
      </a>
      <a
        class="update-team-link"
        href="/teams/2/update"
      >
        Update a Team
      </a>
      <button
        aria-label="DeleteTony as Monk"
        type="button"
        value="delete"
      >
        Delete
      </button>
    </li>
  </ul>
</div>
`;
